Build failure: kdash

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Just build kdash on Linux.
```
nix-build -A kdash
```


### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318039390

### Relevant log output

```console
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_st_init_strend_table_with_size':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:588:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *, st_str_end_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   588 |     str_end_cmp,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:588:5: note: (near initialization for 'hashType.compare')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:550:1: note: 'str_end_cmp' declared here
warning: onig_sys@69.8.1:   550 | str_end_cmp(st_str_end_key* x, st_str_end_key* y)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:589:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   589 |     str_end_hash,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:589:5: note: (near initialization for 'hashType.hash')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:571:1: note: 'str_end_hash' declared here
warning: onig_sys@69.8.1:   571 | str_end_hash(st_str_end_key* x)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_st_init_callout_name_table_with_size':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:678:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *, st_callout_name_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   678 |     callout_name_table_cmp,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:678:5: note: (near initialization for 'hashType.compare')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:637:1: note: 'callout_name_table_cmp' declared here
warning: onig_sys@69.8.1:   637 | callout_name_table_cmp(st_callout_name_key* x, st_callout_name_key* y)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:679:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   679 |     callout_name_table_hash,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:679:5: note: (near initialization for 'hashType.hash')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:660:1: note: 'callout_name_table_hash' declared here
warning: onig_sys@69.8.1:   660 | callout_name_table_hash(st_callout_name_key* x)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'names_clear':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:804:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   804 |     onig_st_foreach(t, i_free_name_entry, 0);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, NameEntry *, void *) {aka int (*)(unsigned char *, NameEntry *, void *)}
warning: onig_sys@69.8.1: In file included from oniguruma/src/regparse.c:37:
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, void *)' {aka 'int (*)(unsigned char *, NameEntry *, void *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:789:1: note: 'i_free_name_entry' declared here
warning: onig_sys@69.8.1:   789 | i_free_name_entry(UChar* key, NameEntry* e, void* arg ARG_UNUSED)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_foreach_name':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:873:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   873 |     onig_st_foreach(t, i_names, (HashDataType )&narg);
warning: onig_sys@69.8.1:       |                        ^~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, NameEntry *, INamesArg *) {aka int (*)(unsigned char *, NameEntry *, INamesArg *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, INamesArg *)' {aka 'int (*)(unsigned char *, NameEntry *, INamesArg *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:846:1: note: 'i_names' declared here
warning: onig_sys@69.8.1:   846 | i_names(UChar* key ARG_UNUSED, NameEntry* e, INamesArg* arg)
warning: onig_sys@69.8.1:       | ^~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_renumber_name_table':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:901:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   901 |     onig_st_foreach(t, i_renumber_name, (HashDataType )map);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, NameEntry *, GroupNumMap *) {aka int (*)(unsigned char *, NameEntry *, GroupNumMap *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, GroupNumMap *)' {aka 'int (*)(unsigned char *, NameEntry *, GroupNumMap *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:879:1: note: 'i_renumber_name' declared here
warning: onig_sys@69.8.1:   879 | i_renumber_name(UChar* key ARG_UNUSED, NameEntry* e, GroupNumMap* map)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'callout_name_table_clear':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1386:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:  1386 |     onig_st_foreach(t, i_free_callout_name_entry, 0);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(st_callout_name_key *, CalloutNameEntry *, void *)
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_callout_name_key *, CalloutNameEntry *, void *)'
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1370:1: note: 'i_free_callout_name_entry' declared here
warning: onig_sys@69.8.1:  1370 | i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'setup_ext_callout_list_values':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1884:56: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:  1884 |     onig_st_foreach((CalloutTagTable *)ext->tag_table, i_callout_callout_list_set,
warning: onig_sys@69.8.1:       |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                                                        |
warning: onig_sys@69.8.1:       |                                                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1866:1: note: 'i_callout_callout_list_set' declared here
warning: onig_sys@69.8.1:  1866 | i_callout_callout_list_set(UChar* key, CalloutTagVal e, void* arg)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'callout_tag_table_clear':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1932:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:  1932 |     onig_st_foreach(t, i_free_callout_tag_entry, 0);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1922:1: note: 'i_free_callout_tag_entry' declared here
warning: onig_sys@69.8.1:  1922 | i_free_callout_tag_entry(UChar* key, CalloutTagVal e, void* arg ARG_UNUSED)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~~
error: failed to run custom build command for `onig_sys v69.8.1`
```

### Additional context

Failed to build the dependent library: `oniguruma`
Related to #475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.1-zen1, NixOS, 26.05 (Yarara), 26.05.20251219.7d853e5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/s1gb3k0sa7q6cal9va8z9anvywrb0p6m-source`

### Notify maintainers

@matthiasbeyer 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/kd/kdash/package.nix
+++ b/pkgs/by-name/kd/kdash/package.nix
@@ -31,6 +31,9 @@ rustPlatform.buildRustPackage rec {
     xorg.xcbutil
   ];
 
+  # Fix for build failure with gcc15
+  env.NIX_CFLAGS_COMPILE = "-std=gnu17";
+
   cargoHash = "sha256-72DuM64wj8WW6soagodOFIeHvVn1CPpb1T3Y7GQYsbs=";
 
   meta = {

